components/DatePicker.vue

```diff
@@ -0,0 +1,74 @@
+<script setup lang="ts">
+import { DatePicker as VCalendarDatePicker } from 'v-calendar';
+import type { DatePickerDate } from 'v-calendar/dist/types/src/use/datePicker';
+import 'v-calendar/dist/style.css';
+
+const today = new Date();
+const minDate = new Date(today).setDate(today.getDate() - 7);
```

Comment:
I'm not too sure where this number 7 comes from. It'd be nice to programmatically calculate this instead of hard-coding this. (Or simply use a library like date-fns)

Reply:
@eunjae-lee Ah, it should've been `6`, but basically it's the earliest date users can go back to given today's date (as in, users can only backtrack and post belated posts within the past week). I'm not too familiar with `date-fns` yet so will peruse the docs to better code this.

Reply:
Oh, so you wanted to subtract by 6 days? `new Date().getTime()` gives you a number (milli seconds) which can be used for calculation.

```
const minDate = new Date(new Date().getTime() - 6*24*60*60*1000)
```

This should be correct 🤔

Reply:
> Oh, so you wanted to subtract by 6 days? `new Date().getTime()` gives you a number (milli seconds) which can be used for calculation.
> 
> ```
> const minDate = new Date(new Date().getTime() - 6*24*60*60*1000)
> ```
> 
> This should be correct 🤔

You know how sometimes you get so stuck in your original flow of thinking it's hard to think of another way (even if it's simpler)? That's what happened to me for this. Yes this is exactly what I want, thank you!!!

Reply:
you've gotta sleep on it 😆